File: Advanced DSA/01. Time Complexity/01. Count Factors - 2/Solution.cpp
int Solution::solve(int A) {
    int factors_count = 0;
    for (int i = 1; i*i <= A; ++i)
        if (A % i == 0)
            factors_count += ((A / i == i) ? 1 : 2);
    return factors_count;
}

File: Advanced DSA/01. Time Complexity/01. Count Factors - 2/Solution.py
class Solution:
    # @param A : integer
    # @return an integer
    def solve(self, A):
        factors_count = 0
        for i in range(1, A+1):
            if i*i > A:
                break
            if A % i == 0:
                factors_count += (1 if A // i == i else 2)
        return factors_count
